src/System.Management.Automation/engine/remoting/commands/CustomShellCommands.cs

```diff
@@ -188,7 +188,7 @@ public sealed class RegisterPSSessionConfigurationCommand : PSSessionConfigurati
                     }}
                }}
                if (([System.Management.Automation.Runspaces.PSSessionConfigurationAccessMode]::Local.Equals($accessMode) -or
-                    ([System.Management.Automation.Runspaces.PSSessionConfigurationAccessMode]::Remote.Equals($accessMode))) -and !$haveDisableACE)
+                    [System.Management.Automation.Runspaces.PSSessionConfigurationAccessMode]::Remote.Equals($accessMode)) -and $haveDisableACE)
                {{
                     # Add network deny ACE for local access or remote access with PSRemoting disabled.
```

Comment:
From the comment below it seems that this is a condition that should be checked...